File: sdk/postgres-sdk/java/example/TestAliyunEmbedding.java
package example;

import com.postgres.AliyunEmbedding;
import com.postgres.PostgresClient;

import java.util.Arrays;
import java.util.List;

public class TestAliyunEmbedding {
    public static void main(String[] args) throws Exception {
        System.out.println("=".repeat(60));
        System.out.println("测试 Java SDK 阿里云百炼 Embedding");
        System.out.println("=".repeat(60));
        
        // 1. 测试直接使用 AliyunEmbedding
        System.out.println("\n1. 测试直接使用 AliyunEmbedding...");
        AliyunEmbedding embedding = new AliyunEmbedding();
        System.out.println("✓ AliyunEmbedding 初始化成功!");
        
        System.out.println("\n" + "=".repeat(60));
        System.out.println("2. 测试单个文本embedding");
        System.out.println("=".repeat(60));
        
        List<String> testTexts = Arrays.asList(
            "Hello world",
            "你好世界",
            "衣服的质量杠杠的",
            "数据处理"
        );
        
        for (String text : testTexts) {
            System.out.println("\n文本: " + text);
            double[] vector = embedding.embedText(text);
            System.out.println("  向量维度: " + vector.length);
            System.out.print("  前5个值: [");
            for (int i = 0; i < 5 && i < vector.length; i++) {
                if (i > 0) System.out.print(", ");
                System.out.printf("%.4f", vector[i]);
            }
            System.out.println("]");
        }
        
        System.out.println("\n" + "=".repeat(60));
        System.out.println("3. 测试 PostgresClient 使用阿里云");
        System.out.println("=".repeat(60));
        
        PostgresClient client = new PostgresClient(true);
        System.out.println("✓ PostgresClient (useAliyun=true) 初始化成功!");
        
        String text = "机器学习";
        System.out.println("\n文本: " + text);
        double[] vector = client.getEmbedding(text);
        System.out.println("  向量维度: " + vector.length);
        System.out.print("  前5个值: [");
        for (int i = 0; i < 5 && i < vector.length; i++) {
            if (i > 0) System.out.print(", ");
            System.out.printf("%.4f", vector[i]);
        }
        System.out.println("]");
        
        System.out.println("\n" + "=".repeat(60));
        System.out.println("✓ 所有测试通过!");
        System.out.println("=".repeat(60));
        System.out.println("\n使用说明:");
        System.out.println("  // 直接使用 AliyunEmbedding");
        System.out.println("  AliyunEmbedding embedding = new AliyunEmbedding();");
        System.out.println("  double[] vector = embedding.embedText(\"你的文本\");");
        System.out.println();
        System.out.println("  // 通过 PostgresClient 使用");
        System.out.println("  PostgresClient client = new PostgresClient(true);");
        System.out.println("  double[] vector = client.getEmbedding(\"你的文本\");");
    }
}

File: sdk/postgres-sdk/go/example/TestAliyunEmbedding.go
package main

import (
	"fmt"
	"log"

	"github.com/example/postgres-sdk-go/postgres"
)

func main() {
	config := postgres.NewConfig("localhost", 5432, "postgres", "postgres", "postgres")
	client := postgres.NewClient(config, true, "", "")

	if err := client.Connect(); err != nil {
		log.Fatalf("Failed to connect: %v", err)
	}
	defer client.Disconnect()

	fmt.Println("Testing Aliyun Embedding...")
	embedding, err := client.GetEmbedding("Hello, world!")
	if err != nil {
		log.Fatalf("Failed to get embedding: %v", err)
	}
	fmt.Printf("Embedding dimension: %d\n", len(embedding))
	fmt.Printf("First 5 values: %v\n", embedding[:5])

	fmt.Println("\nTesting text search...")
	collectionName := "test_aliyun_embedding"

	if err := client.CreateCollection(collectionName, 1024, "IVFFLAT"); err != nil {
		log.Fatalf("Failed to create collection: %v", err)
	}
	defer client.DropCollection(collectionName)

	texts := []string{
		"Hello, world!",
		"这是一个测试",
		"Machine learning is fun",
		"PostgreSQL with pgvector",
	}
	metadata := []map[string]interface{}{
		{"source": "test", "id": 1},
		{"source": "test", "id": 2},
		{"source": "test", "id": 3},
		{"source": "test", "id": 4},
	}

	if err := client.InsertTextWithEmbedding(collectionName, texts, metadata); err != nil {
		log.Fatalf("Failed to insert texts: %v", err)
	}

	results, err := client.SearchByText(collectionName, "Hello", 3, "")
	if err != nil {
		log.Fatalf("Failed to search: %v", err)
	}

	fmt.Println("\nSearch results:")
	for i, result := range results {
		fmt.Printf("%d. ID: %d, Score: %.4f, Metadata: %v\n", i+1, result.ID, result.Score, result.Metadata)
	}

	fmt.Println("\nAll tests passed!")
}
